### PR TITLE
feat: VerifyNonMembership 07-tendermint implementation

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #1291 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/light-clients/07-tendermint/types/client_state_test.go [hunk 1]
```diff
@@ -473,6 +473,238 @@ func (suite *TendermintTestSuite) TestVerifyMembership() {
...
+func (suite *TendermintTestSuite) TestVerifyNonMembership() {
+	var (
+		testingpath         *ibctesting.Path
+		delayTimePeriod     uint64
+		delayBlockPeriod    uint64
+		proofHeight         exported.Height
+		proof               []byte
+		path                []byte
+		invalidClientID     = "09-tendermint"
+		invalidConnectionID = "connection-100"
```
Comment (seantking) [inline modules/light-clients/07-tendermint/types/client_state_test.go:485]: Kinda redundant vars. Open to suggestions for improvements.

**Review by damiannolan — APPROVED** [review]
Nice work, looks pretty clean to me!

Lots of tests, some may be redundant as I think `PacketReceiptAbsence` is the only thing we call `VerifyNonMembership` with atm, but I think its worth having these tests tbh!

LGTM 🚀

File: modules/light-clients/07-tendermint/types/client_state_test.go [hunk 2]
```diff
@@ -473,6 +473,238 @@ func (suite *TendermintTestSuite) TestVerifyMembership() {
...
+		invalidChannelID    = "channel-800"
+		invalidPortID       = "invalid-port"
```
[the remark this answers is not in the document]
Comment (seantking) [reply modules/light-clients/07-tendermint/types/client_state_test.go:487]: Hmm, I think I still prefer vars for readability tbh :-D

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/light-clients/07-tendermint/types/client_state_test.go:487]: Sometimes I find it easier for readability without vars tbh. Depends on the complexity of the tests.
These are probably fine. But my personal choice would be to remove the client/connection id ones

**Pull request merged**